x-pack/plugins/search_indices/public/components/start/create_index.tsx

```diff
@@ -54,6 +54,7 @@ export const CreateIndexForm = ({
 }: CreateIndexFormProps) => {
   const { application } = useKibana().services;
   const [indexNameHasError, setIndexNameHasError] = useState<boolean>(false);
+  const [defaultIndexName] = useState<string>(formState.indexName);
```

Comment:
I think we should track this on the `formState` directly. doing it here has the potential to be inaccurate since this `useState` will be initialized when the `CreateIndexForm` is rendered. So it you change the name, open Code view then come back to UI I _think_ this could have the non-default name. definitely an edge case but I think we can just update `initCreateIndexState()` & `CreateIndexFormState`. 

```
function initCreateIndexState(): CreateIndexFormState {
  const indexName = generateRandomIndexName();
  return {
    indexName,
    defaultIndexName: indexName,
    codingLanguage: getDefaultCodingLanguage(),
  };
}
```

Reply:
Yep formState a nice way of keeping the original value.